### PR TITLE
BP-13544 Add Assign Incident Action Module

### PR DESCRIPTION
**Pull request by JuanDCardozo**

### CODE REVIEW

**Review by cjcrist — COMMENTED** [review]
Added a few comments, but looks good!

File: plugins/modules/assign.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            incident_id=dict(type='str', required=True),
+            assignee_id=dict(type='str', required=True),
+        ),
+        supports_check_mode=True
+    )
+
+    environment_id = module.params['environment_id']
+    api_token = module.params['api_token']
+    incident_id = module.params['incident_id']
+    assignee_id = module.params['assignee_id']
```
[the remark this answers is not in the document]
Comment (JuanDCardozo) [reply plugins/modules/assign.py:76]: Yup!

File: plugins/modules/assign.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            assignee_id=dict(type="str", required=True),
+        ),
+        supports_check_mode=True
+    )
+
+    try:
+        environment_id = module.params["environment_id"]
+        api_token = module.params["api_token"]
+        incident_id = module.params["incident_id"]
+        assignee_id = module.params["assignee_id"]
```
Comment (cjcrist) [inline plugins/modules/assign.py:77]: This wouldn't be an `HTTPError`.  You need to handle the possibility that the user creates a `KeyError` by not assigning a value to one of these params.

**Review by cjcrist — COMMENTED** [review]
1 more update

File: plugins/modules/assign.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        # Prepare the JSON data
+        data = {
+            "assignee": assignee_id
+        }
+
+        # Set the request headers
+        headers = {
+            "Authorization": f"Bearer {api_token}",
+            "Content-Type": "application/json",
```
Comment (cjcrist) [inline plugins/modules/assign.py:88]: You can remove the Content-Type in the header.   It is being set on line 95 with the `json` param.

**Review by cjcrist — APPROVED** [review]
Approved with 1 minor comment.

**Pull request merged**